sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2665,6 +2665,23 @@ object SQLConf {
       .checkValue(_ > 0, "The difference must be positive.")
       .createWithDefault(4)
 
+  val NOT_IN_SUBQUERY_SINGLE_COLUMN_OPTIMIZE_ENABLED =
+    buildConf("spark.sql.notInSubquery.singleColumn.optimize.enabled")
+      .internal()
+      .doc("When true, single column not in subquery execution in BroadcastNestedLoopJoinExec " +
+        "will be optimized from M*N calculation into M*log(N) calculation using HashMap lookup " +
+        "instead of Looping lookup.")
+      .booleanConf
+      .createWithDefault(false)
+
+  val NOT_IN_SUBQUERY_SINGLE_COLUMN_OPTIMIZE_ROW_COUNT_THRESHOLD =
```

Comment:
What should be the relationship of this config vs spark.sql.autoBroadcastJoinThreshold ? Shouldn't the threshold be based on the size of the build size in bytes vs num rows ? I think it is confusing to have two configs for the similar sort of information: How big can the table be.

Reply:
done remove this config and use spark.sql.autoBroadcastJoinThreshold